packages/amplify-graphql-auth-transformer/src/__tests__/resolvers/__snapshots__/subscriptions.test.ts.snap

```diff
@@ -0,0 +1,391 @@
+    #else
+      #set( $ctx.args.filter = { \\"and\\": [ { \\"or\\": $authRuntimeFilter }, $ctx.args.filter ]} )
+    #end
+    #set( $isAuthorized = true )
+  #end
+#end
+#if( !$isAuthorized )
+$util.unauthorized()
+#end
+$util.toJson({\\"version\\":\\"2018-05-29\\",\\"payload\\":{}})
+## [End] Authorization Steps. **"
+`;
+
+exports[`subscriptions generateAuthExpressionForSubscriptions userPools renders for simple userPool auth 1`] = `
+"## [Start] Authorization Steps. **
+$util.qr($ctx.stash.put(\\"hasAuth\\", true))
+#set( $isAuthorized = false )
+#if( $util.authType() == \\"User Pool Authorization\\" )
+  #set( $hasValidOwnerArgument = false )
+  #if( !$isAuthorized )
+    #set( $staticGroupRoles = [{\\"claim\\":\\"cognito:groups\\",\\"entity\\":\\"Admin\\"}] )
+    #foreach( $groupRole in $staticGroupRoles )
+      #set( $groupsInToken = $util.defaultIfNull($ctx.identity.claims.get($groupRole.claim), []) )
+      #if( $groupsInToken.contains($groupRole.entity) )
+        #set( $isAuthorized = true )
+        #break
+      #end
+    #end
+  #end
+  #set( $authRuntimeFilter = [] )
+  #set( $authOwnerRuntimeFilter = [] )
+  #set( $authGroupRuntimeFilter = [] )
+  #set( $ownerClaim0 = $util.defaultIfNull($ctx.identity.claims.get(\\"username\\"), $util.defaultIfNull($ctx.identity.claims.get(\\"cognito:username\\"), null)) )
+  #if( !$util.isNull($ownerClaim0) )
+
+    #set( $ownerClaimsList0 = [] )
+    $util.qr($authOwnerRuntimeFilter.add({ \\"owner\\": { \\"eq\\": $ownerClaim0 } }))
+    #set( $ownerEntity0 = $util.defaultIfNull($ctx.args.owner, null) )
+    #if( !$isAuthorized && !$util.isNullOrEmpty($ownerEntity0) )
+      #if( $ownerEntity0 == $ownerClaim0 || $ownerClaimsList0.contains($ownerEntity0) )
+        #set( $isAuthorized = true )
+        #set( $hasValidOwnerArgument = true )
+      #else
+        $util.unauthorized()
+      #end
+    #end
+  #end
+  #set( $groupClaim1 = $util.defaultIfNull($ctx.identity.claims.get(\\"cognito:groups\\"), []) )
+  #if( $util.isString($groupClaim1) )
+    #if( $util.isList($util.parseJson($groupClaim1)) )
+      #set( $groupClaim1 = $util.parseJson($groupClaim1) )
+    #else
+      #set( $groupClaim1 = [$groupClaim1] )
+    #end
+  #end
+  $util.qr($authGroupRuntimeFilter.add({ \\"group\\": { \\"in\\": $groupClaim1 } }))
+  ## Apply dynamic roles auth if not previously authorized by static groups and owner argument **
+  #if( $authOwnerRuntimeFilter.size() > 0 )
+    $util.qr($authRuntimeFilter.addAll($authOwnerRuntimeFilter))
+  #end
+  #if( $authGroupRuntimeFilter.size() > 0 )
+    $util.qr($authRuntimeFilter.addAll($authGroupRuntimeFilter))
+  #end
+  #if( (!$isAuthorized || $hasValidOwnerArgument) && $authRuntimeFilter.size() > 0 )
+    #if( $util.isNullOrEmpty($ctx.args.filter) )
+      #set( $ctx.args.filter = { \\"or\\": $authRuntimeFilter } )
+    #else
+      #set( $ctx.args.filter = { \\"and\\": [ { \\"or\\": $authRuntimeFilter }, $ctx.args.filter ]} )
+    #end
```

Comment:
If I'm reading this correctly, we will always generate a RTF for owner auth here. 

E.g. `OnCreateTodo(owner: "bob")` will rely on enhanced subscription filtering, even though it can use the (less expensive for the service) basic sub filtering. 

We should only evaluate to RTF when it's necessary:
* `OnCreateTodo(owner: "bob")` -> basic filtering, don't set RTF
* `OnCreateTodo` -> RTF (becase we have to compute the filter in the resolver)
* `OnCreateTodo(owner: "bob", filter: { title: { beginsWith: "a" } })` -> RTF

Same thing with OIDC above.

Reply:
Yeah, agreed on room for the optimization here. Working on separating out that use-case a bit.